frontend/lib/src/components/shared/StreamlitMarkdown/styled-components.ts

```diff
@@ -134,63 +134,59 @@ export const StyledStreamlitMarkdown =
     }
   )
 
-export const StyledLinkIconContainer = styled.div(() => ({
-  position: "relative",
-  left: "calc(-2.5rem - 0.5rem)",
-  width: "calc(100% + 2.5rem + 0.5rem)",
-  display: "flex",
-  alignItems: "center",
-  overflow: "visible",
-  ":hover": {
-    a: {
-      opacity: 1,
-      transform: "scale(1)",
-      transition: "none",
-    },
-  },
-}))
-
 export const StyledLinkIcon = styled.a(({ theme }) => ({
-  position: "absolute",
-  marginRight: "0.5rem",
-
   // center icon
   lineHeight: 0,
   display: "inline-flex",
   alignItems: "center",
   justifyContent: "center",
 
-  // copied from full screen button
-  transform: "scale(0)",
-  transition: "opacity 300ms 150ms, transform 300ms 150ms",
-  opacity: 0,
-  height: "2.5rem",
-  width: "2.5rem",
-  zIndex: theme.zIndices.sidebar + 1,
-  border: "none",
-  backgroundColor: theme.colors.lightenedBg05,
-  borderRadius: "50%",
-
   svg: {
+    // same color as the tooltip-icon
     stroke: theme.colors.fadedText60,
+    strokeWidth: 2.25,
   },
 
   "&:hover svg": {
     stroke: theme.colors.bodyText,
   },
 }))
 
-export const StyledHeaderContainer = styled.div(({ theme }) => ({
+export const StyledHeadingWithActionElements = styled.div(({ theme }) => ({
   "h1, h2, h3, h4, h5, h6, span": {
     scrollMarginTop: theme.spacing.threeXL,
   },
   ...sharedMarkdownStyle(theme),
+
+  // break-word & pretty makes most headings break in a nicer way than break-all while still
+  // preventing overflowing of the container to the side. Long headings without whitespaces or hyphens might still look weird
+  wordBreak: "break-word",
+  textWrap: "pretty",
+
+  // show link-icon when hovering somewhere over the heading
+  [StyledLinkIcon as any]: {
+    visibility: "hidden",
```

Comment:
Any chance we considered disabling this show only on hover behavior for small screen sizes to be more mobile-friendly? We are doing this for the sidebar close button show-on-hover effect.

Reply:
This is how it looks like on mobile (hovered over the second headline so it shows the link icon):

<img width="445" alt="Screenshot 2024-05-06 at 22 02 25" src="https://github.com/streamlit/streamlit/assets/3775781/5e89772d-5160-4ff4-9f28-b35b05f78f3d">

Would you change something based on that?

Reply:
I was just thinking it would be helpful to keep the anchor visible all the time (not just on hover) for smaller screen sizes, but the behavior above is the consistent with our current anchor button, so whichever you think is best!

Reply:
ah gotcha, interesting point! I personally think hiding it looks better, but will loop in @jrieke and Jessi. I might not block merging the PR on it but if they like showing it by default I will create a quick follow-up PR